pubsublite-beam-io/src/main/java/com/google/cloud/pubsublite/beam/PubsubLiteUnboundedReader.java

```diff
@@ -100,9 +114,32 @@ public PubsubLiteUnboundedReader(
                 permanentError = Optional.of(permanentError.orElse(error));
               }
             });
+    this.backlogReader = backlogReader;
+    this.backlogCache =
+        CacheBuilder.newBuilder()
+            .ticker(ticker)
+            .maximumSize(1)
+            .expireAfterWrite(1, TimeUnit.MINUTES)
+            .refreshAfterWrite(10, TimeUnit.SECONDS)
+            .build(
+                new CacheLoader<Object, Long>() {
+                  public Long load(Object val) throws InterruptedException, ExecutionException {
+                    return computeSplitBacklog().get().getMessageBytes();
+                  }
+                });
     this.committerProxy.startAsync().awaitRunning();
   }
 
+  private ApiFuture<ComputeMessageStatsResponse> computeSplitBacklog() {
+    ImmutableMap.Builder<Partition, Offset> builder = ImmutableMap.builder();
+    try (CloseableMonitor.Hold h = monitor.enter()) {
+      subscriberMap.forEach(
+          (partition, subscriberState) ->
+              builder.put(partition, subscriberState.lastDelivered.orElse(Offset.of(0))));
```

Comment:
Oh, yeah, I overlooked that. I've changed this to do the same thing we do when checkpointing, and just ignore partitions in our split for which we've never received a message. 

I guess this could underreport backlog if a partition is somehow stuck, but that's probably fine since scaling up the pipeline wouldn't help in that case anyway